services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_services.py

```diff
@@ -1,20 +1,225 @@
+    summary=(
+        "returns a list of running interactive services filtered by user_id and/or project_id"
+        "both legacy (director-v0) and modern (director-v2)"
+    ),
+)
+async def list_running_dynamic_services(
+    user_id: Optional[UserID] = None,
+    project_id: Optional[ProjectID] = None,
+    director_v0_client: DirectorV0Client = Depends(get_director_v0_client),
+    dynamic_services_settings: DynamicServicesSettings = Depends(get_settings),
+    monitor: DynamicSidecarsMonitor = Depends(get_monitor),
+) -> List[Dict[str, str]]:
+    legacy_running_services: List[
+        DynamicServiceOut
+    ] = await director_v0_client.get_running_services(user_id, project_id)
+
+    get_stack_statuse_tasks: List[DynamicServiceOut] = [
+        monitor.get_stack_status(service["Spec"]["Labels"]["uuid"])
+        for service in await list_dynamic_sidecar_services(
+            dynamic_services_settings.dynamic_sidecar, user_id, project_id
+        )
+    ]
+    modern_running_services: List[Dict[str, str]] = [
+        x.dict(exclude_unset=True)
+        for x in await asyncio.gather(*get_stack_statuse_tasks)
+    ]
+
+    return legacy_running_services + modern_running_services
+
+
+@router.post(
+    "",
+    summary="create & start the dynamic service",
+    status_code=status.HTTP_201_CREATED,
+    response_model=DynamicServiceOut,
+)
+@log_decorator(logger=log)
+async def create_dynamic_service(
+    service: DynamicServiceCreate,
+    x_dynamic_sidecar_request_dns: str = Header(...),
+    x_dynamic_sidecar_request_scheme: str = Header(...),
+    director_v0_client: DirectorV0Client = Depends(get_director_v0_client),
+    dynamic_services_settings: DynamicServicesSettings = Depends(get_settings),
+    monitor: DynamicSidecarsMonitor = Depends(get_monitor),
+) -> DynamicServiceOut:
+    simcore_service: SimcoreService = await director_v0_client.get_service_labels(
+        service=ServiceKeyVersion(key=service.key, version=service.version)
+    )
+    if not simcore_service.needs_dynamic_sidecar:
+        # forward to director-v0
+        redirect_url_with_query = director_v0_client.client.base_url.copy_with(
+            path="/v0/running_interactive_services",
+            params={
+                "user_id": f"{service.user_id}",
+                "project_id": f"{service.project_id}",
+                "service_uuid": f"{service.node_uuid}",
+                "service_key": f"{service.key}",
+                "service_version": f"{service.version}",
+                "service_basepath": f"{service.basepath}",
+            },
+        )
+        log.debug("Redirecting %s", redirect_url_with_query)
+        return RedirectResponse(redirect_url_with_query)
+
+    # if service is already running return the status
+    if await is_dynamic_service_running(
+        dynamic_services_settings.dynamic_sidecar, service.node_uuid
+    ):
+        return await monitor.get_stack_status(str(service.node_uuid))
```

Comment:
I wonder if an error would not be more appropriate here?

Reply:
Not here. This information is sent directly to the frontend and is displayed in the labels. An error would be interpreted like an issue with the system. I think this is the appropriate way to do it.